wolfcrypt/test/test.c

```diff
@@ -26606,80 +26606,90 @@ static int ecc_ctx_kdf_salt_test(WC_RNG* rng, ecc_key* a, ecc_key* b)
     ecEncCtx* bCtx = NULL;
     const byte salt[16] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14,
         15};
-    int ret = 0, aRet = -1, bRet = -1;
-    const char* message = "Hello wolfSSL!";
-    word32 plaintextLen = sizeof(message), encryptLen = 128, decryptLen = 128;
+    int ret = 0;
+    const char message[] = "Hello wolfSSL!";
+    word32 plaintextLen;
+    word32 encryptLen = 128;
+    word32 decryptLen = 128;
 
 #if defined(WOLFSSL_SMALL_STACK) && !defined(WOLFSSL_NO_MALLOC)
     plaintext = XMALLOC(128, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
     encrypted = XMALLOC(128, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
     decrypted = XMALLOC(128, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
 #endif
 
-    ret = aRet = wc_ecc_init(a);
-
-    if (ret == 0)
-        ret = bRet = wc_ecc_init(b);
+    wc_ecc_free(a);
+    wc_ecc_free(b);
 
-    if (ret == 0)
-        ret = wc_ecc_make_key(rng, 32, a);
+    ret = wc_ecc_init(a);
+    if (ret != 0)
+        ret = -10480;
+    if (ret == 0) {
+        ret = wc_ecc_init(b);
+        if (ret != 0)
+            ret = -10481;
+    }
 
+    ret = wc_ecc_make_key(rng, 32, a);
     if (ret == 0)
         ret = wc_ecc_make_key(rng, 32, b);
 
     /* create context */
     if (ret == 0) {
         aCtx = wc_ecc_ctx_new(REQ_RESP_CLIENT, rng);
-
         if (aCtx == NULL)
-            ret = -1;
+            ret = -10470;
     }
 
     if (ret == 0) {
         bCtx = wc_ecc_ctx_new(REQ_RESP_SERVER, rng);
-
         if (bCtx == NULL)
-            ret = -1;
+            ret = -10471;
     }
 
     /* set salt */
-    if (ret == 0)
+    if (ret == 0) {
         ret = wc_ecc_ctx_set_kdf_salt(aCtx, salt, sizeof(salt));
+        if (ret != 0)
+            ret = -10472;
+    }
 
-    if (ret == 0)
+    if (ret == 0) {
         ret = wc_ecc_ctx_set_kdf_salt(bCtx, salt, sizeof(salt));
+        if (ret != 0)
+            ret = 10473;
+    }
 
     XMEMCPY(plaintext, message, XSTRLEN(message));
-
-    while (plaintextLen % AES_BLOCK_SIZE != 0) {
-        plaintextLen++;
-    }
+    plaintextLen = (((word32)XSTRLEN(message) + AES_BLOCK_SIZE - 1) /
+        AES_BLOCK_SIZE) * AES_BLOCK_SIZE;
+    XMEMSET(plaintext + XSTRLEN(message), 0, plaintextLen - XSTRLEN(message));
```

Comment:
sanitizer is not happy with this construction.  probably just do what I did in #6081 -- pre-zero `plaintext` in its entirety, `XMEMSET(plaintext, 0, 128);`  I retested with that change and it was fine.
```
In file included from /usr/include/string.h:535,
                 from ./wolfssl/wolfcrypt/types.h:623,
                 from wolfcrypt/test/test.c:34:
In function ‘memset’,
    inlined from ‘ecc_ctx_kdf_salt_test’ at wolfcrypt/test/test.c:26666:5,
338d8db274 (<106998124+jpbland1@users.noreply.github.com> 2023-02-10 13:05:52 -0500 27333)         ret = ecc_ctx_kdf_salt_test(&rng, userA, userB);
    inlined from ‘ecc_encrypt_test’ at wolfcrypt/test/test.c:27333:15:
/usr/include/bits/string_fortified.h:59:10: error: ‘__builtin_memset’ offset [0, 1] is out of the bounds [0, 0] [-Werror=array-bounds]
   59 |   return __builtin___memset_chk (__dest, __ch, __len,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   60 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/bits/string_fortified.h:59:10: error: ‘__builtin_memset’ offset [0, 1] is out of the bounds [0, 0] [-Werror=array-bounds]
```

found by the `sanitizer-all-asm-smallstack-async` test, i.e. `'--enable-all' '--enable-asynccrypt' '--enable-intelasm' '--enable-sp-math-all' '--enable-smallstack' '--enable-smallstackcache' 'CPPFLAGS=-pedantic' 'CC=gcc' 'CFLAGS=-g -fno-omit-frame-pointer -fsanitize=address,pointer-subtract,leak,undefined,float-cast-overflow,float-divide-by-zero,bounds-strict -fsanitize-recover=all --param=max-vartrack-size=128000000 -march=native' 'LDFLAGS=-g -fno-omit-frame-pointer -fsanitize-recover=all -fsanitize=address,pointer-subtract,leak,undefined,float-cast-overflow,float-divide-by-zero,bounds-strict -fno-sanitize-recover=all '`

the async part has nothing to do with it AFAIK -- all about the smallstack.

Reply:
See: https://github.com/wolfSSL/wolfssl/pull/6081/files#diff-28520d015f1d43f2061e6286aea6efc713b4dba91818eef1206611acb817bd13R26681